sqlglot/dialects/mysql.py

```diff
@@ -189,6 +189,7 @@ class Generator(Generator):
             exp.TryCast: no_trycast_sql,
             exp.DateAdd: _date_add_sql("ADD"),
             exp.DateSub: _date_add_sql("SUB"),
+            exp.DateDiff: lambda self, e: f"DATEDIFF({self.sql(e, 'this')}, {self.sql(e, 'expression')})",
```

Comment:
It's a spelling difference, the default is `DATE_DIFF` but it's spelled `DATEDIFF` in MySQL, Postgres, Snowflake, etc. Maybe we should just change the default to `DATEDIFF` since that seems more common

Reply:
you can use rename_func in this case, and sure you can change the default, but use rename_func helper

Reply:
done!